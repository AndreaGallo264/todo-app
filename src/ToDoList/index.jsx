import React from "react";

//Components
import ToDo from "../ToDo";

const ToDoList = ({ toDoList, handleToggle }) => {
  return (
    <>
      <h1>To do list component</h1>
      <div>
        {toDoList.map((todo) => {
          return <ToDo todo={todo} handleToggle={handleToggle} />;
        })}
      </div>
    </>
  );
};

export default ToDoList;
