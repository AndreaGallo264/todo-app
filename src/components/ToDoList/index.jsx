import React from "react";
import { List, Typography, Grid, Paper, Box } from "@material-ui/core";
import TaskFree from "../../assets/empty.png";
//Components
import ToDo from "../ToDo";

const ToDoList = ({ list }) => {
  return (
    <Grid container direction="column">
      <Box display="flex" flexDirection="column" alignItems="center">
        <Typography variant="h3" color="primary" align="center" paragraph>
          Your tasks
        </Typography>
        <Paper elevation={3} style={{ width: "80%" }}>
          {list && (
            <List>
              {list.length > 0 ? (
                <>
                  {list.map((todo) => {
                    return <ToDo todo={todo} key={todo.id} />;
                  })}
                </>
              ) : (
                <Box display="flex" flexDirection="column" alignItems="center">
                  <img src={TaskFree} alt="No more tasks" width="200" />
                  <Typography
                    variant="h3"
                    color="secondary"
                    align="center"
                    paragraph
                  >
                    You have no tasks left!
                  </Typography>
                </Box>
              )}
            </List>
          )}
        </Paper>
      </Box>
    </Grid>
  );
};

export default ToDoList;
