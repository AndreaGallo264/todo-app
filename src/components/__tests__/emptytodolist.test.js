import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import ToDoList from "../ToDoList";

describe("Empty ToDoList component", () => {
  test("Matches snapshot", () => {
    const mockStore = configureMockStore();
    const store = mockStore({ list: [], lastId: 0 });
    const tree = renderer
      .create(
        <Provider store={store}>
          <ToDoList />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
