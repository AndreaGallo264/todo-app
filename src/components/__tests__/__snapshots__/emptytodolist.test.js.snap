// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty ToDoList component Matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <p
      className="MuiTypography-root MuiTypography-h3 MuiTypography-colorPrimary MuiTypography-paragraph MuiTypography-alignCenter"
    >
      Your tasks
    </p>
    <div
      className="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
      style={
        Object {
          "width": "80%",
        }
      }
    >
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          className="MuiBox-root MuiBox-root-2"
        >
          <img
            alt="No more tasks"
            src="empty.png"
            width="200"
          />
          <p
            className="MuiTypography-root MuiTypography-h3 MuiTypography-colorSecondary MuiTypography-paragraph MuiTypography-alignCenter"
          >
            You have no tasks left!
          </p>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
